model-optimizer/extensions/front/tf/RetinaNetFilteredDetectionsReplacement.py

```diff
@@ -246,9 +246,18 @@ def generate_sub_graph(self, graph: Graph, match: SubgraphMatch):
                                                                    applied_width_height_regressions_node)
 
         detection_output_op = DetectionOutput(graph, match.custom_replacement_desc.custom_attributes)
+        # get nms from the original network
+        iou_threshold = None
+        for node_name in match.matched_nodes_names():
+            if Node(graph, node_name)['op'] == 'NonMaxSuppression':
+                iou_threshold = Node(graph, node_name).in_node(3).value
+                break
+        if iou_threshold is not None:
+            raise Error('During {} `iou_threshold` was not retrieved from RetinaNet subgraph'.format(self.replacement_id))
```

Comment:
assert iou_threshold is not None, 'During {} `iou_threshold` was not retrieved from RetinaNet subgraph'.format(self.replacement_id)

Reply:
Usually asserts are our internal check for consistency. While 'raise Error' is a more user-friendly error that we want to present to the user. So in this particular case I would recommend to leave 'raise Error'